src/components/Counter.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export class Counter extends React.Component {
+  state = {
+    counter: 0,
+  }
+
+  addOne = () => {
+    this.setState(state => ({
+      counter: state.counter + 1,
+    }));
+  };
+
+  add100 = () => {
+    this.setState(state => ({
+      counter: state.counter + 100,
+    }));
+  };
+
+  increase = () => {
+    this.setState((state) => {
+      if (state.counter % 5 === 0) {
+        this.addOne();
+        this.add100();
+      } else {
+        this.addOne();
+      }
+    });
+  };
```

Comment:
Я думаю, что это условие можно немного упростить
```suggestion
  increase = () => {
      this.addOne();
      if (this.state.counter % 5 === 0) {
        this.add100();
      }
  };
```

Reply:
totally agree, thank you